### PR TITLE
Fix bounds for embedded lists in QP

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Fixes bounds computations for embedded lists for QP sidebar. `!= None` is not a valid filtering approach for multikey searches

## How is this patch tested? If it is not, please explain why.

Added lightning test coverage

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced numeral formatting to improve error handling for "NaN" outputs.
	- Updated `RangeSlider` to provide more informative feedback when no bounds are available.
  
- **Bug Fixes**
	- Improved handling of special float values (inf, nan) in query responses.
	- Enhanced error handling and control flow in view loading and filtering processes.

- **Tests**
	- Updated test cases to reflect changes in data handling and expected outcomes for various data types.

### CODE REVIEW

**Review by CamronStaley — APPROVED** [review]

File: app/packages/core/src/components/Filters/NumericFieldFilter/state.ts [hunk 1]
```diff
@@ -25,14 +25,16 @@ export const hasDefaultRange = selectorFamily({
     },
 });
 
-export const hasNonfinites = selectorFamily({
-  key: "hasNonfinites",
+export const nonfinitesText = selectorFamily({
```
Comment (kaixi-wang) [inline app/packages/core/src/components/Filters/NumericFieldFilter/state.ts:28]: is nonfinite supposed to mean not finite or not infinite?

File: fiftyone/server/lightning.py [hunk 2]
```diff
@@ -317,7 +317,7 @@ async def _do_async_query(
     filter: t.Optional[t.Mapping[str, str]],
 ):
     if isinstance(query, DistinctQuery):
-        if query.has_list and not query.filters:
+        if query.has_list:
             return await _do_distinct_query(collection, query, filter)
```
Comment (kaixi-wang) [inline fiftyone/server/lightning.py:321]: not sure about mongo, but distinct scans in most dbs are really expensive and slow... have you confirmed that relying more on distinct improves performance?

Comment (benjaminpkane) [reply fiftyone/server/lightning.py:321]: A distinct scan with high cardinality is not good, yes. I am not aware of an aggregation pipeline that can return the first N sorted results using a multikey index, though. But we should investigate more

File: app/packages/core/src/components/Common/utils.tsx [hunk 3]
```diff
@@ -68,13 +68,14 @@ export const getFormatter = (fieldType: string, timeZone: string, bounds) => {
...
 
-      return numeral(v).format(
+      const str = numeral(v).format(
         [INT_FIELD, FRAME_NUMBER_FIELD, FRAME_SUPPORT_FIELD].includes(fieldType)
           ? "0a"
           : bounds[1] - bounds[0] < 0.1
           ? "0.0000a"
           : "0.00a"
       );
+      return str === "NaN" ? v.toString() : str;
```
Comment (minhtuev) [inline app/packages/core/src/components/Common/utils.tsx:78]: if str is not a number (null?), we just show it as null?

**Review by minhtuev — APPROVED** [review]
Small comment, otherwise LGTM

**Pull request merged**